fastapi_storages/s3.py

```diff
@@ -110,5 +112,27 @@ def write(self, file: BinaryIO, name: str) -> str:
 
         file.seek(0, 0)
         key = self.get_name(name)
+
+        if not self.overwrite_existing_files:
+            key = self.rename_file(key)
+
         self._bucket.upload_fileobj(file, key, ExtraArgs={"ACL": self.AWS_DEFAULT_ACL})
         return key
+    
+    def rename_file(self, filename: str) -> str:
+        path = Path(filename)
+        base = path.parent / path.stem
+        objects = list(self._bucket.objects.filter(Prefix=str(base)))
```

Comment:
I strongly suggest to use `head_object` for this: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/head_object.html

This will be a lightweight HEAD request to check if the key exists.

Reply:
I have two options here:
1. To use for loop as I did in `FileStorage` to find suitable file name.
2. Get all files with given prefix, sort them, find the number of file, and use next available number. (Currently using this one). 

1st is multiple requests (1 request for cheking one possible name, so if I have `filename_5.txt` it will be 6 requests)
2nd is one bigger request, independent from filenames.

Reply:
Yeah I think the the first one makes more sense.
The second one has a higher risk that if the prefix is too common the list might be too big and pagination might be needed.